integration/util/registry.go

```diff
@@ -0,0 +1,197 @@
+	body   string
+
+	user string
+	pass string
+
+	delete bool
+}
+
+type BootstrapConfiguration struct {
+	LogFile             string   `json:"logFile"`
+	PostBootstrapFile   string   `json:"postBootstrapFile"`
+	PostBootstrapScript []string `json:"postBootstrapScript"`
+	CaCert              string   `json:"caCert"`
+	Address             string   `json:"address"`
+	TenantID            string   `json:"tenantId"`
+	DeviceID            string   `json:"deviceId"`
+	AuthID              string   `json:"authId"`
+	Password            string   `json:"password"`
+}
+
+type ConnectorConfiguration struct {
+	CaCert   string `json:"caCert"`
+	LogFile  string `json:"logFile"`
+	Address  string `json:"address"`
+	TenantID string `json:"tenantId"`
+	DeviceID string `json:"deviceId"`
+	AuthID   string `json:"authId"`
+	Password string `json:"password"`
+}
+
+// CreateDeviceResources creates device resources.
+func CreateDeviceResources(newDeviceId, tenantID, policyID, password, registryAPI,
+	registryAPIUsername, registryAPIPassword string, cfg *TestConfiguration) []*Resource {
+
+	devicePath := tenantID + "/" + newDeviceId
+	return []*Resource{
+		&Resource{
+			url:    registryAPI + "/devices/" + devicePath,
+			method: http.MethodPost,
+			body:   `{"authorities":["auto-provisioning-enabled"]}`,
+			user:   registryAPIUsername,
+			pass:   registryAPIPassword,
+			delete: true},
+		&Resource{
+			url:    registryAPI + "/credentials/" + devicePath,
+			method: http.MethodPut,
+			body:   getCredentialsBody(strings.ReplaceAll(newDeviceId, ":", "_"), password),
+			user:   registryAPIUsername,
+			pass:   registryAPIPassword},
+		&Resource{
+			url:    GetThingURL(cfg.DigitalTwinAPIAddress, newDeviceId),
+			method: http.MethodPut,
+			body:   fmt.Sprintf(`{"policyId": "%s"}`, policyID),
+			user:   cfg.DigitalTwinAPIUsername,
+			pass:   cfg.DigitalTwinAPIPassword,
+			delete: true},
+	}
+}
+
+func getCredentialsBody(authID, pass string) string {
+	type pwdPlain struct {
+		PwdPlain string `json:"pwd-plain"`
+	}
+
+	type authStruct struct {
+		TypeStr string     `json:"type"`
+		AuthId  string     `json:"auth-id"`
+		Secrets []pwdPlain `json:"secrets"`
+	}
+	auth := authStruct{"hashed-password", authID, []pwdPlain{pwdPlain{pass}}}
+
+	data, _ := json.MarshalIndent([]authStruct{auth}, "", "\t")
+	return string(data)
+}
+
+// DeleteResources deletes all given resources and all related devices.
+func DeleteResources(cfg *TestConfiguration, resources []*Resource, deviceId, url, user, pass string) error {
+	var errStrings []string
+	if err := deleteRelatedDevices(cfg, deviceId, url, user, pass); err != nil {
+		errStrings = append(errStrings, err.Error())
+	}
+
+	// Delete in reverse order of creation
+	for i := len(resources) - 1; i >= 0; i-- {
+		r := resources[i]
+
+		if !r.delete {
+			continue
+		}
+
+		if _, err := SendDeviceRegistryRequest(nil, http.MethodDelete, r.url, r.user, r.pass); err != nil {
+			errStrings = append(errStrings, err.Error())
+		}
```

Comment:
we can simplify this a bit
```suggestion
		r := resources[i]
 		if r.delete {
 		    if _, err := SendDeviceRegistryRequest(nil, http.MethodDelete, r.url, r.user, r.pass); err != nil {
 			    errStrings = append(errStrings, err.Error())
 		    }
 	       }
 ```